src/Build/Evaluation/Evaluator.cs

```diff
@@ -1334,21 +1341,24 @@ private void EvaluatePropertyElement(ProjectPropertyElement propertyElement)
 
                 _expander.UsedUninitializedProperties.CurrentlyEvaluatingPropertyElementName = null;
 
-                P predecessor = _data.GetProperty(propertyElement.Name);
-
-                P property = _data.SetProperty(propertyElement, evaluatedValue, predecessor);
+                P property = _data.SetProperty(propertyElement, evaluatedValue);
 
-                if (predecessor != null)
+                if (Traits.Instance.LogPropertyTracking == 0)
                 {
-                    LogPropertyReassignment(predecessor, property, propertyElement.Location.LocationString);
+                    P predecessor = _data.GetProperty(propertyElement.Name);
```

Comment:
Perhaps I'm confused, but should this not go above line 1344?

Reply:
It should. Refactoring.